packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -346,44 +348,42 @@ export type Metadata = {
 };
 
 /**
- * Controller that passively polls on a set interval for NFT auto detection
+ * Controller that passively detects nfts for a user address
  */
-export class NftDetectionController extends StaticIntervalPollingController<
+export class NftDetectionController extends BaseController<
```

Comment:
@sahar-fehri what do you think about making this a service instead as it's stateless ?

Reply:
Resolving this now because we discussed that we can iterate in the future to extract api interactions into a separate service.

Reply:
For future reference, I believe [modeling stateless services as non-controllers](https://github.com/MetaMask/decisions/blob/main/decisions/core/0001-messaging-non-controllers.md#decision-outcome) is a separate concern from [extracting external API integrations](https://github.com/MetaMask/decisions/blob/a0ae0e0e63e15e848e5138c214042f9880719598/decisions/core/0001-external-api-integrations.md). 

We probably want to create a ticket for the first task if it's out of scope for this PR.